app/src/main/java/de/danoeh/antennapod/activity/BugReportActivity.java

```diff
@@ -67,42 +72,60 @@ protected void onCreate(Bundle savedInstanceState) {
             clipboard.setPrimaryClip(clip);
             Snackbar.make(findViewById(android.R.id.content), R.string.copied_to_clipboard, Snackbar.LENGTH_SHORT).show();
         });
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.bug_report_options, menu);
+        return super.onCreateOptionsMenu(menu);
+    }
 
-        findViewById(R.id.btn_export_logcat).setOnClickListener(v -> {
-            try {
-                File filename = new File(UserPreferences.getDataFolder(null), "full-logs.txt");
-                filename.createNewFile();
-                String cmd = "logcat -d -f " + filename.getAbsolutePath();
-                Runtime.getRuntime().exec(cmd);
-                //share file
+    @Override
+    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
+        if (item.getItemId() == R.id.export_logcat) {
+            AlertDialog.Builder alertBuilder = new AlertDialog.Builder(this);
+            alertBuilder.setTitle("Warning!").setMessage("Logs may contain sensitive information");
+            alertBuilder.setPositiveButton("Continue", (dialog, which) -> {
                 try {
-                    Intent i = new Intent(Intent.ACTION_SEND);
-                    i.setType("text/*");
-                    String authString = getString(de.danoeh.antennapod.core.R.string.provider_authority);
-                    Uri fileUri = FileProvider.getUriForFile(this, authString, filename);
-                    i.putExtra(Intent.EXTRA_STREAM,  fileUri);
-                    i.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
-                    if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.KITKAT) {
-                        PackageManager pm = getPackageManager();
-                        List<ResolveInfo> resInfos = pm.queryIntentActivities(i, PackageManager.MATCH_DEFAULT_ONLY);
-                        for (ResolveInfo resolveInfo : resInfos) {
-                            String packageName = resolveInfo.activityInfo.packageName;
-                            grantUriPermission(packageName, fileUri, Intent.FLAG_GRANT_READ_URI_PERMISSION);
+                    File filename = new File(UserPreferences.getDataFolder(null), "full-logs.txt");
```

Comment:
Could you please move this export code to a new function? This is now a really long method and would be easier to read if split in two.